driver/src/price_finding/stablex_naive_solver.rs

```diff
@@ -0,0 +1,208 @@
+use crate::price_finding::error::PriceFindingError;
+use crate::price_finding::PriceFinding;
+use crate::util;
+use dfusion_core::models::{AccountState, Order, Solution};
+use std::cmp::Ordering;
+
+/// A naive solver for the stableX contract. It works almost identically to the
+/// Snapp naive solver except it has a few extra requirements and it doesn't try
+/// to be as optimal.
+///
+/// The solver works by finding the first pair of orders which matching sell and
+/// buy tokens, where one of the traded tokens is the fee token, whose users have
+/// enough balance to fully fill, and that has a viable solution.
+///
+/// The solution is computed by taking the price of the larger order and and
+/// completely filling the smaller order. A large order is defined by having
+/// more non-fee token volume. Account balances are checked for each solution.
+pub struct StableXNaiveSolver;
+
+impl PriceFinding for StableXNaiveSolver {
+    fn find_prices(
+        &self,
+        orders: &[Order],
+        accounts: &AccountState,
+    ) -> Result<Solution, PriceFindingError> {
+        Ok(ipairs(orders)
+            .filter(|&(i, j)| {
+                orders[i].buy_token == orders[j].sell_token
+                    && orders[i].sell_token == orders[j].buy_token
+            })
+            .filter(|&(i, _)| orders[i].buy_token == 0 || orders[i].sell_token == 0)
+            .filter(|&(i, j)| {
+                has_enough_balance(&orders[i], accounts) && has_enough_balance(&orders[j], accounts)
+            })
+            .filter_map(|(i, j)| compute_solution(orders, i, j))
+            .nth(0)
```

Comment:
what the heck is this? Does this have something to do with taking the first match!

Reply:
Rust iterators don't have a `first()`. `nth(0)` is the 0th element in the iterator and is equivalent to `first()`.